### PR TITLE
Enable asan on clang

### PR DESCRIPTION
**Pull request by williamcroberts**
Fixes: #387 
Fixes: #375

### CODE REVIEW

**Comment by williamcroberts** [comment]
@martinezjavier

File: lib/tpm2_alg_util.h [hunk 1]
```diff
@@ -136,4 +136,14 @@ struct tpm2_pcr_digest_spec {
 bool pcr_parse_digest_list(char **argv, int len,
         tpm2_pcr_digest_spec *digest_spec);
 
+/**
+ * Retrieves the size of a hash in bytes for a given hash
+ * algorithm or 0 if unkown/not found.
+ * @param id
```
Comment (martinezjavier) [inline lib/tpm2_alg_util.h]: typo: unknown

File: lib/tpm_session.c [hunk 2]
```diff
@@ -242,63 +138,59 @@ TPM_RC tpm_session_start_auth_with_params(TSS2_SYS_CONTEXT *sapi_context, SESSIO
...
     {
-        DeleteSession( *session );
+        memcpy( &(*session)->salt.b, &salt->b, sizeof(salt->b));
     }
-    return( rval );
+
+    if( (*session)->bind == TPM_RH_NULL )
+        (*session)->authValueBind.t.size = 0;
+
+
```
Comment (martinezjavier) [inline lib/tpm_session.c]: Extra blank line here.

File: test/system/test_all.sh [hunk 3]
```diff
@@ -86,7 +86,19 @@ test_wrapper test_tpm2_verifysignature.sh
...
+# 1. Detecting a leak that (maybe) doesn't exist.
+#    OpenSSL is hard...
+# 2. The suppresion option via ASAN_OPTIONS doesn't
+#    exist for 3.6.
+# TODO When this is fixed, remove it.
+# Bug: https://github.com/01org/tpm2-tools/issues/390
+if [ "$ASAN_ENABLED" != "true" ]; then
+  test_wrapper test_tpm2_getmanufec.sh
+fi
+
```
Comment (martinezjavier) [inline test/system/test_all.sh]: Not that nice, but hopefully it can be eventually sort it out.

**Comment by martinezjavier** [comment]
@williamcroberts patches looks good to me. So I'm OK with this after the build failure reported by travis is fixed.

**Comment by williamcroberts** [comment]
Weird, it's a link error:
```
/home/travis/build/01org/tpm2-tools/build/../lib/tpm2_nv_util.c:486: undefined reference to `Tss2_Sys_NV_ReadPublic'
```
I'll have to dig into this tomorrow, but we can't dynamically link this into the static lib in a portable fashion. Perhaps require that helper routines for TSS code be static inlined in the header...

**Comment by williamcroberts** [comment]
Blocked on failing tpm2-abrmd build:
https://github.com/01org/tpm2-abrmd/issues/126

**Comment by williamcroberts** [comment]
I don't like inlining these things like in:
https://github.com/01org/tpm2-tools/pull/391/commits/3507b0c0badc2cb833f6d75ea62645494b9acf4e

I'll see if I can switch to clang 3.6 since locally that is what I have and it works.

**Comment by williamcroberts** [comment]
Stupid comment + close button, this is now on #396. After one does a force-push it cannot re-open a closed PR.

**Pull request closed without merging**